_posts/2024-03-11-release-0_5_0.md

```diff
@@ -0,0 +1,40 @@
+---
+layout: post
+title:  "Kroxylicious release 0.5.0"
+date:   2024-03-10 00:00:00 +0000
+author: "Robert Young"
+author_url: "https://www.github.com/robobario"
+categories:  [releases, kroxylicious, record-encryption]
+---
+
+The Kroxylicious project is very pleased to announce the release of [Kroxylicious 0.5.0](https://github.com/kroxylicious/kroxylicious/releases/tag/v0.5.0).
+
+### Record Encryption
+
+This release was focused on refining [Record Encryption](/use-cases) to be ready for experimentation in secure environments, adding:
+* Enable users to configure Hashicorp Vault TLS, providing custom keystore and truststore
+* Enable users to supply the Hashicorp Vault token via file rather than having to inject it into the proxy configuration
+* Include Record Encryption filter (and other Kroxylicious project supported filters) in the binary distribution
+* Release a Docker image `quay.io/kroxylicious/kroxylicious` also containing the supported filters
+* Internal refactoring to control the usage of key material, preparing to support alternate Ciphers and configurable Additional Authenticated Data
+
+The protocol for immutable encrypted data written to the broker is now aligned with our initial design and we guarantee data encrypted with version 0.5.0 
+of the Filter will be decryptable forever (assuming the keys stored in the KMS are available forever for decryption).
+
+> **_NOTE:_**: Until now we have used the term Envelope Encryption when describing the Filter. Going forward we will use the name Record Encryption 
+to better describe the granularity that encryption is happening at. Encryption is done per [record](https://kafka.apache.org/documentation/#record).
+Envelope Encryption is how we encrypt those records. Record Encryption is how we implement Encryption-At-Rest.
+
+### Other Improvements
+
+* Support for Apache Kafka 3.7.0 API additions (Kroxylicious is forwards-compatible, but must be modified if we wish to intercept new
+APIs or fields added to existing APIs)
```

Comment:
I think we want to expand this a little.

Something like 
```suggestion
* Support for Apache Kafka 3.7.0 API additions (Kroxylicious is version agnostic and will automatically interfaces between all broker and client versions, but must be updated to enable the interception of new
APIs or fields added to existing APIs)
```

Reply:
I'm finding the second version harder to follow, personally. I wonder if it could be worded a bit differently, something like
```suggestion
* Support for Apache Kafka 3.7.0 API additions (Kroxylicious is version agnostic and can interface with all current broker and client versions, but must be updated to intercept new
APIs or fields added to existing APIs)
```

Reply:
What I was trying to express was that you can drop a new client or broker version either side of Kroxylicious and it will just work, there is no need for co-ordinated upgrades. So really it would be `current and future` broker and client versions`. But it does need upgraded to expose or intercept the new APIs.

Though I fully accept that could be far too much nuance for the release notes.